WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -166,6 +167,38 @@ def do_import
     redirect_to competition_registrations_import_url(competition)
   end
 
+  def add
+    @competition = competition_from_params
+  end
+
+  def do_add
+    @competition = competition_from_params
+    if @competition.registration_full?
```

Comment:
Is there a race condition here if there are two registrations happening simultaneously where only 1 should happen, but they both do? I'm not too familiar with the competition limit enforcement code, so this problem very well may exist in other places. If the problem does exist in other places, I'm ok with waiting to take care of this another time (maybe as part of https://github.com/thewca/worldcubeassociation.org/issues/3451?)

Reply:
There most likely is, although in this case I believe it's unlikely enough (definitely less likely than in case of a registration process with multiple people registering). Definitely something we can work on separately.